course_info/views.py

```diff
@@ -170,6 +187,33 @@ def editor(request):
     course_context = _course_context(request, _ORDERED_FIELD_NAMES, True,
                                      course_instance_id=course_instance_id)
     course_context['launch_presentation_return_url'] = \
-            request.POST.get('launch_presentation_return_url')
+        request.POST.get('launch_presentation_return_url')
     course_context['textarea_fields'] = ['description', 'notes']
     return render(request, 'course_info/editor.html', course_context)
+
+
+def sort_and_format_instructor_display(course_instructor_list):
+    """
+    Returns a sorted  string  sorted by  the instructors by role_id, seniority sort, last name
+    and then formatted such that it appears as a comma delimited String with an 'and' before that last user.
+    # Eg: User1, User2 and User3.
+    """
+    instructor_display = ''
+    # Note:  when seniority_sort is null, it was getting precedence over 1(eg: null, 1, 2).  So in such cases,
+    # it is being set to a large number (picked 100) so it is lower in the sorting order. [1, 2, ...null(set to 100)]
+    course_instructor_list.sort(key=lambda x: (x.get('role')['role_id'],
+                                               100 if x.get('seniority_sort') is None else x.get('seniority_sort'),
+                                               x.get('profile')['name_last']))
+
+    for person in course_instructor_list:
+        instructor_display += person.get('profile')['name_first']+' '+person.get('profile')['name_last']+', '
+
+    # Replace the last ',' with a period and the one before with an and
+    instructor_display = replace_right(instructor_display, ', ', '.', 1)
+    instructor_display = replace_right(instructor_display, ', ', ' and ', 1)
```

Comment:
Took a while to figure out what this was trying to do, and why.  I'm still not sure it does the right thing if we've got 1 or 2 people.

Does it make more sense to write a `display_name()` method that operates on a `person`, then do something like (replacing my `map` with something more iterative if you're not someone who likes `map`):

```
course_instructor_names = map(display_name, course_instructor_list)
num_instructors = len(course_instructor_list)
if num_instructors == 1:
    return course_instructor_names[0] + '.'
elif num_instructors == 2:
    return ' and '.join(course_instructor_names) + '.'
elif num_instructors > 2:
    # add the last instructor name after an 'and'
    # s/, and/and/ if you're a non-oxford-comma-using savage
    return ', '.join(course_instructor_names[:-1]) + ', and' + course_instructor_names[-1] + '.'
else:
    # num_instructors == 0
    return ''